### PR TITLE
Feat/filter layers by tags

### PR DESCRIPTION
**Pull request by bob-voorhoede**
- multiselect component for tags
- filter list of layers by tag selection
- small refactor for less code and more logical code location

### CODE REVIEW

File: src/store/modules/data/index.js [hunk 1]
```diff
@@ -1,41 +1,59 @@
 import configRepo from '~/repo/configRepo'
-import { VALID_PLATFORMS } from '~/lib/constants'
+import { flattenLayers, getLayersTags } from '~/lib/layer-helpers'
 
 export default {
   namespaced: true,
 
   state: () => ({
-    layers: [],
+    // originalLayers: [],
```
Comment (bob-voorhoede) [inline src/store/modules/data/index.js:8]: I keep this around just until we're sure we don't need it. Check definitively after we have rendered the layers on the map, and figured out what happens if you switch between 'all layers' and 'filtered flattened layers' by using the multiselect tag selector

File: src/views/Layers.vue [hunk 2]
```diff
@@ -1,36 +1,62 @@
...
+      :layers="layersForTree"
       @active-layers-change="onActiveLayerChange"
       @layer-sorting-change="onLayerSortingChange"
     />
   </v-card>
 </template>
 
 <script>
-  import { mapActions, mapState } from 'vuex'
+  import { mapActions, mapGetters } from 'vuex'
```
Comment (bob-voorhoede) [inline src/views/Layers.vue:23]: I never ever use state directly in templates (you can accidentally mutate state, never a getter)

File: src/views/Layers.vue [hunk 3]
```diff
@@ -1,36 +1,62 @@
...
+
     computed: {
-      ...mapState('data', [ 'layers' ]),
+      ...mapGetters('data', [
+        'displayLayers',
+        'flattenedLayers',
+        'layerTags',
+      ]),
+
+      layersForTree() {
```
Comment (bob-voorhoede) [inline src/views/Layers.vue:44]: If it turns out to be troublesome to replace the input for the layer tree, there's also a filter option in the `v-treeview` component, but it's currently not exposed by the `layer-list-controls` component

File: src/store/modules/app/index.js [hunk 4]
```diff
@@ -25,4 +16,14 @@ export default {
       state.appNavigationOpen = open
     },
   },
+
+  actions: {
```
Comment (vvandijck) [inline src/store/modules/app/index.js:20]: Why did you move this code?

**Review by vvandijck — APPROVED** [review]
